File: my_alpha_player/android/src/main/kotlin/com/example/my_alpha_player/MyAlphaPlayerPlugin.kt
package com.example.my_alpha_player

import io.flutter.embedding.engine.plugins.FlutterPlugin

/** MyAlphaPlayerPlugin */
class MyAlphaPlayerPlugin : FlutterPlugin {

    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        // 核心代码在这里！
        // 当插件连接到 Flutter 引擎时，我们注册视图工厂
        flutterPluginBinding.platformViewRegistry.registerViewFactory(
            "com.example.live/alpha_player", // 必须和 Dart 代码里的 viewType 一模一样
            NativeAlphaPlayerFactory(flutterPluginBinding.binaryMessenger)
        )
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        // 插件卸载时，通常不需要做特殊清理，除非你需要解绑 channel
    }
}

File: my_alpha_player/ios/Classes/MyAlphaPlayerPlugin.swift
import Flutter
import UIKit

public class MyAlphaPlayerPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "my_alpha_player", binaryMessenger: registrar.messenger())
    let instance = MyAlphaPlayerPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)

    // 注意：这里的 viewId 必须和 Dart 端的 viewType 一致
    let factory = NativeAlphaPlayerFactory(messenger: registrar.messenger())
    registrar.register(factory, withId: "com.example.live/alpha_player")
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPlatformVersion":
      result("iOS " + UIDevice.current.systemVersion)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
